spec/helpers/admin/claims_helper_spec.rb

```diff
@@ -173,4 +173,29 @@
       it { is_expected.to be_nil }
     end
   end
+
+  describe "#matching_attributes" do
+    let(:first_claim) {
+      build(
+        :claim,
+        teacher_reference_number: "0902344",
+        national_insurance_number: "QQ891011C",
+        email_address: "genghis.khan@mongol-empire.com",
+        bank_account_number: "34682151",
+        bank_sort_code: "972654",
+        building_society_roll_number: "123456789/ABCD",
+      )
+    }
+    let(:second_claim) { build(:claim, :submitted, teacher_reference_number: first_claim.teacher_reference_number, national_insurance_number: first_claim.national_insurance_number, building_society_roll_number: first_claim.building_society_roll_number) }
+    subject { helper.matching_attributes(first_claim, second_claim) }
+
+    it "returns the humanised names of the matching attributes" do
+      expect(subject).to eq(["Building society roll number", "National insurance number", "Teacher reference number"])
+    end
+
+    it "does not consider nil building society roll number to be a match" do
+      [first_claim, second_claim].each { |claim| claim.update!(building_society_roll_number: nil) }
```

Comment:
Could we avoid database writes with this?

```suggestion
      [first_claim, second_claim].each { |claim| claim.building_society_roll_number =  ni) }
```

Reply:
Good idea; updated.